src/hipscat/pixel_math/partition_stats.py

```diff
@@ -0,0 +1,150 @@
+"""Utilities for generating and manipulating object count histograms"""
+
+import healpy as hp
+import numpy as np
+import pandas as pd
+
+
+def empty_histogram(highest_order):
+    """Use numpy to create an histogram array with the right shape, filled with zeros.
+    Args:
+        highest_order (int): the highest healpix order (e.g. 0-10)
+    Returns:
+        one-dimensional numpy array of long integers, where the length is equal to
+        the number of pixels in a healpix map of target order, and all values are set to 0.
+    """
+    return np.zeros(hp.order2npix(highest_order), dtype=np.ulonglong)
+
+
+def generate_histogram(
+    data: pd.DataFrame,
+    highest_order,
+    ra_column="ra",
+    dec_column="dec",
+):
+    """Generate a histogram of counts for objects found in `data`
+
+    Args:
+        data (`obj`:DataFrame): tabular object data
+        highest_order (int):  the highest healpix order (e.g. 0-10)
+        ra_column (str): where in the input to find the celestial coordinate, right ascension
+        dec_column (str): where in the input to find the celestial coordinate, declination
+    Returns:
+        one-dimensional numpy array of long integers where the value at each index corresponds
+        to the number of objects found at the healpix pixel.
+    Raises:
+        ValueError: if the `ra_column` or `dec_column` cannot be found in the input file.
+    """
+    histo = empty_histogram(highest_order)
```

Comment:
I'm a big fan of avoiding abbreviations for names, is there any reasons not to use `histogram` here?

Reply:
Totally agree.